### ISSUE TITLE
Package autoloads?

### ISSUE THREAD

**Issue by proofit404** [renzmann/treesit-auto#44]
Some users do not use `use-package`.

That's mean they have to `(require 'treesit-auto)` before they could call `(global-treesit-auto-mode)` in the configuration file.

To avoid this situation package could specify special auto load pragma. For example, like [this](https://github.com/bbatsov/projectile/blob/271007c6611fcb08ddd326d7de9727c2ad5ef265/projectile.el#L2440).

Usually, it's good tone to set it for all interactive commands defined in the package.

Is there any specific technical reason not to set auto load comments?

**Comment by renzmann** [comment]
I did try this, but my skill with elisp wasn't quite good enough to overcome `cl-lib` failing to load properly: https://github.com/renzmann/treesit-auto/issues/36#issuecomment-1474875674 

If you are familiar with the proper way to get `;;;###autoload` working without any errors on startup I'd love the PR for it!

**Comment by noctuid** [comment]
No, I didn't add an autoload cookie for the global minor mode in that PR.  The last sentence in that PR explains why an autoload cookie will not work with how the current code is setup.  Basically, the problem is that the :predicate depends on code which will not be pulled into/available in the autoload file, which is why there are errors.
